### PR TITLE
Add readme instructions for DSPO.

### PR DESCRIPTION
**Pull request by HumairAK**
## Description
Add a more detailed readme for the repository frontpage

## How Has This Been Tested?
Follow and replicate the instructions

## Merge criteria:
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by HumairAK** [comment]
See frontpage of my branch to see a preview: https://github.com/HumairAK/data-science-pipelines-operator/tree/add_readme

File: README.md [hunk 1]
```diff
@@ -1,51 +1,307 @@
...
+# Create the namespace if it doesn't already exist
+oc new-project ${DSPO_NS}
+```
+
+Now we will navigate to the DSPO manifests then build and deploy them to this namespace.
+
 ```bash
-oc new-project data-science-pipelines-operator
-cd ${REPO}/config/default
+cd ${WORKGING_DIR}/config/base
```
Comment (gregsheremeta) [inline README.md:98]:
```suggestion
cd ${WORKING_DIR}/config/base
```

File: README.md [hunk 2]
```diff
@@ -1,51 +1,307 @@
...
+
+Once all pods are ready, we can proceed to deploying the first Data Science Pipelines (DSP) instance. Instructions
+[here](#deploy-dsp-instance).
+
+## Deploy the Operator standalone
+
+First clone this repository: 
+
+```bash
+WORKGING_DIR=$(mktemp -d)
```
Comment (gregsheremeta) [inline README.md:80]:
```suggestion
WORKING_DIR=$(mktemp -d)
```

File: README.md [hunk 3]
```diff
@@ -1,51 +1,307 @@
...
+git clone https://github.com/opendatahub-io/data-science-pipelines-operator.git ${WORKGING_DIR}
```
Comment (gregsheremeta) [inline README.md:81]:
```suggestion
git clone https://github.com/opendatahub-io/data-science-pipelines-operator.git ${WORKING_DIR}
```

File: README.md [hunk 4]
```diff
@@ -1,51 +1,307 @@
...
+```
+If you already have the repository cloned, set `WORKGING_DIR=` to its absolute location accordingly.
```
Comment (gregsheremeta) [inline README.md:83]:
```suggestion
If you already have the repository cloned, set `WORKING_DIR=` to its absolute location accordingly.
```

File: README.md [hunk 5]
```diff
@@ -1,51 +1,307 @@
...
 oc new-project ${DSP_Namespace}
-cd ${REPO}/config/samples
+```
+
+DSPO introduces a new custom resource to your cluster called `DSPipeline`. This resource is where you configure your DSP 
+components, your DB and Storage configurations, and so forth. For now, we'll use the sample one provided, feel free to 
+inspect this sample resource to see other configurable options.
+
+```bash
+cd ${WORKGING_DIR}/config/samples
```
Comment (gregsheremeta) [inline README.md:126]:
```suggestion
cd ${WORKING_DIR}/config/samples
```

File: README.md [hunk 6]
```diff
@@ -1,51 +1,307 @@
...
+### Deploy another DSP instance 
+
+You can use the DSPO to deploy multiple `DSPipeline` instances in different OpenShift namespaces, for example earlier 
+we deployed a `DSPipeline` resource named `sample`. We can use this again and deploy it to a different namespace:
 
 ```bash
 DSP_Namespace_2=test-ds-project-2
 oc new-project ${DSP_Namespace_2}
-cd ${REPO}/config/samples
+cd ${WORKGING_DIR}/config/samples
```
Comment (gregsheremeta) [inline README.md:146]:
```suggestion
cd ${WORKING_DIR}/config/samples
```

File: README.md [hunk 7]
```diff
@@ -1,51 +1,307 @@
...
+## Using the API
+
+In the previous step we submitted a generated `Pipeline` yaml via the GUI. We can also submit the `Pipeline` code 
+directly either locally or via a notebook.
+
+You can find the `Pipeline` code example [here][flipcoin code example]. We can submit this to the DSP API Server by 
+including this code in the following Python script:
+
+```bash
+cd ${WORKGING_DIR}/docs/example_pipelines
```
Comment (gregsheremeta) [inline README.md:225]:
```suggestion
cd ${WORKING_DIR}/docs/example_pipelines
```

File: README.md [hunk 8]
```diff
@@ -1,51 +1,307 @@
...
+KFDEF_NAME=data-science-pipelines-operator
+oc delete kfdef ${KFDEF_NAME} -n ${DSPO_NS}
+```
+
+## Cleanup Standalone Installation
+
+To clean up standalone DSPO deployment:
+
+```bash
+# WORKGING_DIR must be the root of this repository's clone
```
Comment (gregsheremeta) [inline README.md:296]:
```suggestion
# WORKING_DIR must be the root of this repository's clone
```

File: README.md [hunk 9]
```diff
@@ -1,51 +1,307 @@
...
+cd ${WORKGING_DIR}/config/base
```
Comment (gregsheremeta) [inline README.md:297]:
```suggestion
cd ${WORKING_DIR}/config/base
```

File: README.md [hunk 10]
```diff
@@ -1,51 +1,307 @@
...
+oc new-project ${DSPO_NS}
+```
+
+Now we will navigate to the DSPO manifests then build and deploy them to this namespace.
+
 ```bash
-oc new-project data-science-pipelines-operator
-cd ${REPO}/config/default
+cd ${WORKGING_DIR}/config/base
 kustomize build . | oc apply -f -
```
Comment (gregsheremeta) [inline README.md:99]: should we use `make` instead? `make deploy`? not sure if it needs to be tweaked though.

```
.PHONY: deploy
deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
	$(KUSTOMIZE) build config/default | kubectl apply -f -

```

Comment (HumairAK) [reply README.md]: great idea, I updated the makefile and instructions accordingly

**Comment by gregsheremeta** [comment]
I'd prefer if the Makefile changes were extracted to a separate commit. If you don't feel like doing it, lmk and I'll lgtm it anyway

otherwise looks great

**Comment by HumairAK** [comment]
> I'd prefer if the Makefile changes were extracted to a separate commit. If you don't feel like doing it, lmk and I'll lgtm it anyway 

commits have been split up, lmk if that works

**Review by rimolive — APPROVED** [review]

**Pull request merged**